src/ICSharpCode.SharpZipLib/BZip2/BZip2.cs

```diff
@@ -40,6 +41,16 @@ public static void Decompress(Stream inStream, Stream outStream, bool isStreamOw
 			}
 		}
 
+		/// <summary>
+		/// Decompress the <paramref name="inStream">input</paramref> writing
+		/// uncompressed data to the <paramref name="outStream">output stream</paramref>
+		/// </summary>
+		/// <param name="inStream">The readable stream containing data to decompress.</param>
+		/// <param name="outStream">The output stream to receive the decompressed data.</param>
+		/// <param name="isStreamOwner">Both streams are closed on completion if true.</param>
+		public static void Decompress(Stream inStream, Stream outStream, bool isStreamOwner)
+			=> Decompress(inStream, outStream, isStreamOwner, true);
```

Comment:
```suggestion
			=> Decompress(inStream, outStream, isStreamOwner, inStream.CanSeek);
```

Reply:
I'm not certain if there aren't any streams that support seeking but don't allow access to their length, but maybe that's too much of an unusual situation to worry about, and as long as the functionality can be disabled that should cover the situation if needed.

Reply:
I don't know if seeking can even be achieved if you don't allow Length (since the SeekOrigin parameter of Seek call can be set to "End".
The Stream documentation basically thinks of them as mutually inclusive:
https://docs.microsoft.com/en-us/dotnet/standard/io/?view=netframework-4.8#streams

Reply:
That's why I said too unusual to worry about (it might not make sense, but that doesn't guarantee it won't happen!).
Not a reason to avoid making the change though